### PR TITLE
1.39.0 release post

### PR DESCRIPTION
**Pull request by Centril**
r? @rust-lang/release @rust-lang/core

Thanks to @Mark-Simulacrum, @pietroalbini, and @tmandry for proof reading.

### CODE REVIEW

File: posts/2019-11-07-Rust-1.39.0.md [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+
+## What's in 1.39.0 stable
+
+The highlights of Rust 1.39.0 include `async`/`.await`, shared references to by-move bindings in `match` guards, and attributes on function parameters. Also, see the [detailed release notes][notes] for additional information.
+
+### The `.await` is over, `async fn`s are here
+
+[rel-1360]: https://blog.rust-lang.org/2019/07/04/Rust-1.36.0.html#the-future-is-here
+[`Future`]: https://doc.rust-lang.org/nightly/std/future/trait.Future.html
+[niko-post-async]: TODO
```
Comment (Centril) [inline posts/2019-11-07-Rust-1.39.0.md]: @nikomatsakis This one's for you. :)

File: posts/2019-11-07-Rust-1.39.0.md [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+### Other changes
+
+[relnotes-cargo]: https://github.com/rust-lang/cargo/blob/master/CHANGELOG.md#cargo-139-2019-11-07
+[relnotes-clippy]: https://github.com/rust-lang/rust-clippy/blob/master/CHANGELOG.md#rust-139
+[compat-notes]: https://github.com/rust-lang/rust/blob/stable/RELEASES.md#compatibility-notes
+
+There are other changes in the Rust 1.39.0 release: check out what changed in [Rust][notes], [Cargo][relnotes-cargo], and [Clippy][relnotes-clippy].
+
+Please also see the [compatability notes][compat-notes] to check if you're affected by those changes.
```
Comment (jonas-schievink) [inline posts/2019-11-07-Rust-1.39.0.md]:
```suggestion
Please also see the [compatibility notes][compat-notes] to check if you're affected by those changes.
```

File: posts/2019-11-07-Rust-1.39.0.md [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+[err-2018]: https://github.com/rust-lang/rust/pull/63565
+[err-2015]: https://github.com/rust-lang/rust/pull/64221
+[rip-ast-borrowck]: https://github.com/rust-lang/rust/pull/64790
+[niko-blog-nll]: https://blog.rust-lang.org/2019/11/01/nll-hard-errors.html
+
+In the 1.35.0 release, [we announced][rel-1350] that NLL had come to Rust 2015 after first being released for Rust 2018 in [1.31][rel-1310].
+
+As noted in the 1.35.0 release, the old borrow checker had some bugs which would allow memory unsafety. These bugs were fixed by the NLL borrow checker. As these fixes broke some stable code, we decided to gradually phase in the errors by checking if the old borrow checker would accept the program and the NLL checker would reject it. If so, the errors would instead become warnings.
+
+With Rust 1.39.0, these warnings are now [errors on Rust 2018][err-2018].
```
Comment (jonas-schievink) [inline posts/2019-11-07-Rust-1.39.0.md]:
```suggestion
With Rust 1.39.0, these warnings are now [errors in Rust 2018][err-2018].
```

File: posts/2019-11-07-Rust-1.39.0.md [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+
+Previously in Rust 1.36.0, [we announced][rel-1360] that the [`Future`] trait is here. Back then, we noted that:
+
+> With this stabilization, we hope to give important crates, libraries, and the ecosystem time to prepare for `async` / `.await`, which we'll tell you more about in the future.
+
+A promise made is a promise kept. So in Rust 1.39.0, we are pleased to announce that `async` / `.await` is stabilized! Concretely, this means that you can define `async` functions and blocks and `.await` them.
+
+An `async` function, which you can introduce by writing `async fn` instead of `fn`, does nothing other than to return a `Future` when called. This `Future` is a suspended computation which you can drive to completion by `.await`ing it. Besides `async fn`, `async { ... }` and `async move { ... }` blocks, which act like closures, can be used to define "async literals".
+
+For more on the release of `async` / `.await`, read [Niko Matsakis's blog post].
```
Comment (funkill) [inline posts/2019-11-07-Rust-1.39.0.md]: @Centril `[Niko Matsakis's blog post][niko-post-async]` here?

Comment (Centril) [reply posts/2019-11-07-Rust-1.39.0.md]: Good catch!
```suggestion
For more on the release of `async` / `.await`, read [Niko Matsakis's blog post][niko-post-async].
```

File: posts/2019-11-07-Rust-1.39.0.md (hunk 1 above)
Comment (Centril) [reply posts/2019-11-07-Rust-1.39.0.md]:
```suggestion
[niko-post-async]: https://blog.rust-lang.org/2019/11/07/Async-await-stable.html
```

**Pull request merged**